frontend/packages/console-shared/src/components/status/status-components.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import * as History from 'history';
+import { Link } from 'react-router-dom';
+import classNames from 'classnames';
+import { Popover, PopoverPosition } from '@patternfly/react-core';
+import { Button } from 'patternfly-react';
+import { ErrorCircleOIcon, InProgressIcon, OkIcon, InfoAltIcon } from '@patternfly/react-icons';
+import {
+  global_danger_color_100 as dangerColor,
+  global_success_color_200 as successColor,
+} from '@patternfly/react-tokens';
+
+import { CamelCaseWrap } from '@console/internal/components/utils';
+import { DASH } from '../../constants';
+
+type StatusIconAndTextProps = {
+  icon?: React.ReactElement;
+  title?: string;
+  spin?: boolean;
+};
+
+export const StatusIconAndText: React.FC<StatusIconAndTextProps> = ({ icon, title, spin }) => {
+  if (!title) {
+    return <>{DASH}</>;
+  }
+
+  return (
+    <>
+      {icon &&
+        React.cloneElement(icon, {
```

Comment:
Nit: I think we should avoid `React.cloneElement` if possible.

Did you consider using an `Icon` prop that references a React component type, instead of its instance?

For example: `Icon && <Icon className={...} otherProp={...} />`

Reply:
@christianvogt suggested the same, that solution prevents ability to pass custom props to the icon. We would have to introduce additional iconProps prop to satisfy that. Maybe passing additional props is YAGNI but I would rather keep that option available to provide a way to pass in custom className or id which IMHO can be a valid use case.